ec/ecresource/trafficfilterresource/flatteners.go

```diff
@@ -18,66 +18,66 @@
 package trafficfilterresource
 
 import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
 	"github.com/elastic/cloud-sdk-go/pkg/models"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-func modelToState(d *schema.ResourceData, res *models.TrafficFilterRulesetInfo) error {
-	if err := d.Set("name", *res.Name); err != nil {
-		return err
-	}
-
-	if err := d.Set("region", *res.Region); err != nil {
-		return err
-	}
+func modelToState(ctx context.Context, res *models.TrafficFilterRulesetInfo, state *modelV0) diag.Diagnostics {
+	var diags diag.Diagnostics
 
-	if err := d.Set("type", *res.Type); err != nil {
-		return err
-	}
+	state.Name = types.String{Value: *res.Name}
+	state.Region = types.String{Value: *res.Region}
+	state.Type = types.String{Value: *res.Type}
+	state.IncludeByDefault = types.Bool{Value: *res.IncludeByDefault}
 
-	if err := d.Set("rule", flattenRules(res.Rules)); err != nil {
-		return err
-	}
+	diags.Append(flattenRules(ctx, res.Rules, &state.Rule)...)
 
-	if err := d.Set("include_by_default", res.IncludeByDefault); err != nil {
-		return err
+	if res.Description == "" {
+		state.Description = types.String{Null: true}
+	} else {
+		state.Description = types.String{Value: res.Description}
 	}
 
-	if res.Description != "" {
-		if err := d.Set("description", res.Description); err != nil {
-			return err
-		}
-	}
-
-	return nil
+	return diags
 }
 
-func flattenRules(rules []*models.TrafficFilterRule) *schema.Set {
-	result := schema.NewSet(trafficFilterRuleHash, []interface{}{})
+func flattenRules(ctx context.Context, rules []*models.TrafficFilterRule, target interface{}) diag.Diagnostics {
```

Comment:
Would be nicer to return a typed model rather than assigning to an interface pointer.

Reply:
@pascal-hofmann ?